app/controllers/application_controller.rb

```diff
@@ -1,10 +1,15 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :create_cart
 
   protected
 
   def configure_permitted_parameters
     devise_parameter_sanitizer.permit(:sign_up, keys: [:name])
   end
+
+  def create_cart
+    session[:cart_id] = ShoppingCart.create.id unless session[:cart_id]
```

Comment:
```
session[:card_id] ||= ShoppingCart.create.id
# is equivalent to 
session[:card_id] = session[:cart_id] || ShoppingCart.create.id
```

This means if `session[:cart_id]` evaluates to `nil` or `false` we create a new `ShoppingCart` and get its `id` assigned to `session[:cart_id]`

Every other call to this method will not override the value of `session[:cart_id]` unless it was cleared somewhere else in the app (ie: shopping cart destroyed)